presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -294,6 +296,71 @@ public void testCreateTableNonExistentPartitionColumns()
     }
 
     @Test
+    public void createPartitionedBucketedTableAs()
+            throws Exception
+    {
+        createPartitionedBucketedTableAs(HiveStorageFormat.RCBINARY);
+    }
+
+    public void createPartitionedBucketedTableAs(HiveStorageFormat storageFormat)
+            throws Exception
+    {
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE test_create_partitioned_bucketed_table_as " +
+                "WITH (" +
+                "format = '" + storageFormat + "', " +
+                "partitioned_by = ARRAY[ 'orderstatus' ], " +
+                "clustered_by = ARRAY[ 'custkey' ], " +
+                "bucket_count = 11 " +
+                ") " +
+                "AS " +
+                "SELECT custkey, comment, orderstatus " +
+                "FROM tpch.tiny.orders";
+
+        assertUpdate(
+                getSession().withSystemProperty("task_writer_count", "3"),
+                createTable,
+                "SELECT count(*) from orders");
+
+        TableMetadata tableMetadata = getTableMetadata("test_create_partitioned_bucketed_table_as");
+        assertEquals(tableMetadata.getMetadata().getProperties().get(STORAGE_FORMAT_PROPERTY), storageFormat);
+
+        List<String> partitionedBy = ImmutableList.of("orderstatus");
+        assertEquals(tableMetadata.getMetadata().getProperties().get(PARTITIONED_BY_PROPERTY), partitionedBy);
+        for (ColumnMetadata columnMetadata : tableMetadata.getColumns()) {
+            boolean partitionKey = partitionedBy.contains(columnMetadata.getName());
+            assertEquals(columnMetadata.getComment(), annotateColumnComment(null, partitionKey));
+        }
+
+        assertEquals(tableMetadata.getMetadata().getProperties().get(CLUSTERED_BY_PROPERTY), ImmutableList.of("custkey"));
+        assertEquals(tableMetadata.getMetadata().getProperties().get(BUCKET_COUNT_PROPERTY), 11);
+
+        List<HivePartition> partitions = getPartitions("test_create_partitioned_bucketed_table_as");
+        assertEquals(partitions.size(), 3);
+
+        // Hive will reorder the partition keys to the end
+        assertQuery("SELECT * from test_create_partitioned_bucketed_table_as", "SELECT custkey, comment, orderstatus FROM orders");
+
+        try {
+            assertUpdate("INSERT INTO test_create_partitioned_bucketed_table_as VALUES (1, 'comment', 'O')", 1);
```

Comment:
Add a test that does a successful insert into a new partition and read some data from it.